vms/evm/subscriber.go

```diff
@@ -154,25 +157,37 @@ func (s *subscriber) ProcessFromHeight(height *big.Int) error {
 		return err
 	}
 
-	// Cap the number of blocks to process to MaxBlocksToProcess
-	toBlock := big.NewInt(0).SetUint64(latestBlock)
-	if height.Cmp(big.NewInt(0).Sub(toBlock, big.NewInt(MaxBlocksToProcess))) < 0 {
-		s.logger.Warn(
-			fmt.Sprintf("Requested to process too many blocks. Processing only the most recent %d blocks", MaxBlocksToProcess),
-			zap.String("requestedBlockHeight", height.String()),
-			zap.String("latestBlockHeight", toBlock.String()),
-			zap.String("chainID", s.chainID.String()),
-		)
-		height = big.NewInt(0).Add(toBlock, big.NewInt(-MaxBlocksToProcess))
+	bigLatestBlock := big.NewInt(0).SetUint64(latestBlock)
+
+	for fromBlock := height; fromBlock.Cmp(bigLatestBlock) <= 0; /*see post statement in body*/ {
```

Comment:
Since `height` is a `*big.Int`, incrementing `fromBlock` will also increment `height`. This isn't a problem now, but could cause a bug in the future if we expand this function to use `height` after this loop. We should initialize the loop using `fromBlock := new(big.Int).Set(height)

I also think having the post statement in the loop body is error prone, as there's nothing stopping somebody from adding code below the post statement. One workaround would be to put the loop body in an anonymous function and execute the post statement in a `defer`, guaranteeing that it's the last statement executed in the body. We'd want to comment that clearly though. Neither way is particularly clean, so I leave it up to you to decide.

Reply:
Good call on the pointer side effect, I'm still getting used to Go land, so this was a great lesson, thank you.

Regarding the error prone for loop, thank you also for pushing on that. What I submitted here was my second iteration, and that was better than the first, but I still wasn't completely happy with it. Your workaround suggestion was good food for thought; I didn't take it, but it made me find a way to do it that _is_ particularly clean (though with a tiny bit of computational inefficiency (calling the `toBlock()` function variable twice per loop iteration), but I thought the tradeoff for cleanliness over maximum efficiency was a good one). I revamped this loop in https://github.com/ava-labs/awm-relayer/pull/89/commits/c4807558f8e8a15905d1854a03bb8660ea2370ba . let me know what you think.